.github/workflows/test.yml

```diff
@@ -0,0 +1,77 @@
+name: test
+on:
+  push:
+    branches:
+      - 'develop'
+      - 'trunk'
+  pull_request:
+
+concurrency:
+  group: branch-node-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      # Test node 16/18 on Ubuntu
+      # Test node 16 on macOS/Windows
+      # Enable linting only for node 16 + ubuntu
+      matrix:
+        os: [ubuntu-latest]
+        node: [14, 16]
+        include:
+          - os: macos-latest
+            node: 16
+          - os: windows-latest
+            node: 16
+          - node: 16
+            is_base_node_version: true
+          - os: ubuntu-latest
+            is_base_os_version: true
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 30
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: env
```

Comment:
What's this doing?

Reply:
Building up env variables for the restore keys below